test/test_main.py

```diff
@@ -143,42 +143,38 @@ def _check(self, charm_class, *, extra_environ=None, **kwargs):
                             fh.write(b'name: test')
                         mock_charmdir.return_value = tmpdirname
 
-                        with warnings.catch_warnings(record=True) as warnings_cm:
-                            main(charm_class, **kwargs)
-
-        return warnings_cm
+                        main(charm_class, **kwargs)
 
     def test_init_signature_passthrough(self):
         class MyCharm(CharmBase):
 
             def __init__(self, *args):
                 super().__init__(*args)
 
-        warn_cm = self._check(MyCharm)
-        self.assertFalse(warn_cm)
+        with warnings.catch_warnings(record=True) as warn_cm:
+            self._check(MyCharm)
+        self.assertEqual(warn_cm, [])
 
     def test_init_signature_both_arguments(self):
         class MyCharm(CharmBase):
 
             def __init__(self, framework, somekey):
                 super().__init__(framework, somekey)
 
-        warn_cm = self._check(MyCharm)
-        self.assertEqual(len(warn_cm), 1)
-        (warn,) = warn_cm
-        self.assertTrue(issubclass(warn.category, DeprecationWarning))
-        self.assertEqual(str(warn.message), (
-            "the second argument, 'key', has been deprecated and will be removed "
-            "after the 0.7 release"))
+        msg = ("the second argument, 'key', has been deprecated and will be removed "
+               "after the 0.7 release")
```

Comment:
This seems like a strange artefact from the past - do we need to remove or update this message?

Reply:
I have the feeling we should just remove Key and get rid of the deprecation warning.

Reply:
Good idea -- done.